xarray/backends/api.py

```diff
@@ -827,6 +828,14 @@ def open_mfdataset(
         - 'override': if indexes are of same size, rewrite indexes to be
           those of the first object with that dimension. Indexes for the same
           dimension must have the same size in all objects.
+    attrs_file : str, optional
+        Path of the file used to read global attributes from.
+        By default global attributes are read from the first file provided.
+        Wildcard matches are sorted by filename. Nested lists are flattened
+        using a dictionary. For Python 3.6 and higher, global attributes are
+        read from the first file of the first list (of the first list etc.).
+        For Python 3.5 or lower, global attributes are read from an
+        arbitrary file.
```

Comment:
Actually it looks like we only currently [explicitly support](https://github.com/pydata/xarray/blob/f2b2f9f62ea0f1020262a7ff563bfe74258ffaa1/setup.py#L24) python 3.6 & 3.7, so this can be shortened to simply 
```
By default global attributes are read from the first file provided, with wildcard matches sorted by filename.
```

Reply:
So... In this case should we just keep the file index option after all? For nested list that would be the index in the flattened list.

Reply:
No, the reason we wanted to avoid the integer index option wasn't because unordered dictionaries made it non-deterministic, but because it's a less clear API for users who aren't required to understand how the list flattening works under the hood. Also, it's not clear what passing an integer index should mean if `combine='by_coords'`.

By keeping only the string specifier, the behaviour is totally clear for users without any knowledge of what happens within either of the `combine` functions.

Reply:
No, the reason we wanted to avoid the integer index option wasn't because unordered dictionaries made it non-deterministic, but because it's a less clear API for users who aren't required to understand how the list flattening works under the hood. Also, it's not clear what passing an integer index should mean if `combine='by_coords'`.

By keeping only the string specifier, the behaviour is totally clear for users without any knowledge of what happens within either of the combine functions.

Reply:
@TomNicholas Agreed. I simplified the docstring as suggested above (aa62e18) and mark this conversation resolved.